distributed/semaphore.py

```diff
@@ -141,6 +153,10 @@ async def acquire(self, comm=None, name=None, timeout=None, lease_id=None):
             w = _Watch(timeout)
             w.start()
 
+            # We create a new uuid in case of multiple acquire retries with the same `lease_id`
```

Comment:
I would actually not introduce a new UUID here. On the one side it introduces even more complexity to the setup and on the other side the retries are more or less an implementation detail we should not necessarily expose via the prometheus metrics. Best case, we'd expose a counter of how often something was retried but I'm not sure if this is valuable since I'd expect this to be an edge case and we have logging info for this as well

Reply:
My rationale was that when this edge case occurs, we don't want this metric to say that a pending_lease is pending for a shorter amount of time than is the actual case. To clarify, you consider this too improbable to be worth implementing?

Reply:
Setting consistency and complexity arguments aside for a moment, I would argue that you can achieve an identical behaviour with a simple counter

```
self.metrics["time_to_acquire_lease"]["pending_leases"] += 1

# do stuff

self.metrics["time_to_acquire_lease"]["pending_leases"] -= 1
self.metrics["time_to_acquire_lease"][name].update(w.elapsed())
```